From 05de024179a49feeb8bab63d27a599d1ad26407d Mon Sep 17 00:00:00 2001
From: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Date: Wed, 10 Oct 2018 11:43:17 -0700
Subject: [PATCH 0462/1852] xilinx: v4l: vipp: Avoid decrementing refcount of a
 node twice

commit 7d1d2b09389ed395bc4152cd376ccf6eb52b44a8 from
https://github.com/Xilinx/linux-xlnx.git

In the current implementation, driver is calling of_graph_get_next_endpoint()
and of_node_put() functions to get next endpoint and decrement refcount of a
node. As the former function is already decrementing refcount, there is no need
of calling the later function which decrements the refcount again.

This patch avoids calling of_node_put() function in driver, as the same has been
taken care in of_graph_get_next_endpoint() implementation.

Signed-off-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-vipp.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-vipp.c b/drivers/media/platform/xilinx/xilinx-vipp.c
index d834e448bbc5..99926d2d363c 100644
--- a/drivers/media/platform/xilinx/xilinx-vipp.c
+++ b/drivers/media/platform/xilinx/xilinx-vipp.c
@@ -176,7 +176,6 @@ static int xvip_graph_build_one(struct xvip_composite_device *xdev,
 		}
 	}
 
-	fwnode_handle_put(ep);
 	return ret;
 }
 
@@ -319,7 +318,6 @@ static int xvip_graph_build_dma(struct xvip_composite_device *xdev)
 		}
 	}
 
-	of_node_put(ep);
 	return ret;
 }
 
-- 
2.31.1

